### DISCUSSION TITLE
GZipMiddleware appears to not honor minimize_size

### DISCUSSION THREAD

**Discussion by flashgorman-chorus** [encode/starlette#1253] [Q&A] [answered]
I have installed `GZipMiddleware` with a `minimize_size` set to 1024.  When I make a request that returns a response smaller than 1024, I expect the response will not be compressed and the `Content-Encoding` header will not be set to `gzip`.

My testing, however, shows that the responses are always compressed and `Content-Encoding` is always set to `gzip`.

Debugging the code, I see `send_with_gzip` is called but `message["more_body"]` is set to `True` the first time through, no matter what, from `testclient.py`:

```python
    async def stream_response(self, send: Send) -> None:
        await send(
            {
                "type": "http.response.start",
                "status": self.status_code,
                "headers": self.raw_headers,
            }
        )
        async for chunk in self.body_iterator:
            if not isinstance(chunk, bytes):
                chunk = chunk.encode(self.charset)
            await send({"type": "http.response.body", "body": chunk, "more_body": True})

        await send({"type": "http.response.body", "body": b"", "more_body": False})
```

As a result, the key block of code in `gzip.py` that I expected to enter is skipped:

```python
if len(body) < self.minimum_size and not more_body:
    # Don't apply GZip to small outgoing responses.
    await self.send(self.initial_message)
    await self.send(message)
```

Since `more_body` is `True`, it never enters this block.  Perhaps this is just an aberration of the test client?  Or I am doing something wrong?

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/encode/starlette/issues/2003 from this discussion to make an investigation. Thanks. :+1:
